### PR TITLE
Remove internal use of `gitOpts` struct from FluxAddonClient

### PR DESCRIPTION
**Pull request by danbudris**
*Issue #, if available:*

*Description of changes:*
removing internal use of the unneeded `gitOpts` struct from the FluxAddonClient. The whole options argument will be removed in favor of directly injecting dependencies in an upcoming pr.

this is an intermediate step to factoring out the `git.Provider` interface into two parts, a GitClient (responsible for local repo actions) and a GitProvider (responsible for actions on a remote provider), which will be generated by the factory and injected directly into the FluxAddonClient (along with the writer) rather than using the intermediate `gitOpts` struct.

*Testing (if applicable):*
`make`, created a cluster with gitops, upgrade it, and deleted it.
created a cluster w/out gitops.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/addonmanager/addonclients/fluxaddonclient.go [hunk 1]
```diff
@@ -95,9 +96,13 @@ func NewGitOptions(ctx context.Context, cluster *v1alpha1.Cluster, fluxConfig *v
 }
 
 func NewFluxAddonClient(flux Flux, gitOpts *GitOptions) *FluxAddonClient {
+	if gitOpts == nil {
```
Comment (danbudris) [inline pkg/addonmanager/addonclients/fluxaddonclient.go:99]: this is an intermediate step to replicate the existing behavior which will be removed once we completely remove the gitOpts and pass in individual dependencies; a different method for determining wether GitOps install should be skipped will be put in place.

Comment (danbudris) [reply pkg/addonmanager/addonclients/fluxaddonclient.go:99]: by 'existing behavior', which check if the 'gitOpts' is nil to determine wether ot install flux or not; we're now just checking wether FluxAddonClient itself is nil

**Review by jiayiwang7 — APPROVED** [review]
/lgtm

File: pkg/dependencies/factory_test.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ func TestFactoryBuildWithMultipleDependencies(t *testing.T) {
 	tt.Expect(deps.Bootstrapper).NotTo(BeNil())
 	tt.Expect(deps.ClusterManager).NotTo(BeNil())
 	tt.Expect(deps.Provider).NotTo(BeNil())
-	tt.Expect(deps.FluxAddonClient).NotTo(BeNil())
+	tt.Expect(deps.FluxAddonClient).To(BeNil())
```
Comment (g-gaston) [inline pkg/dependencies/factory_test.go:82]: why this change? the factory is not changing
